optuna/samplers/_nsga2/sampler.py

```diff
@@ -154,18 +162,36 @@ def sample_relative(
         generation = parent_generation + 1
         study._storage.set_trial_system_attr(trial_id, _GENERATION_KEY, generation)
 
+        dominates = _dominates if self._constraints_func is None else _constrained_dominates
         if parent_generation >= 0:
-            p0 = self._select_parent(study, parent_population)
+            # crossover
             if self._rng.rand() < self._crossover_prob:
-                p1 = self._select_parent(
-                    study, [t for t in parent_population if t._trial_id != p0._trial_id]
+                child = crossover(
+                    self._crossover_name,
+                    study,
+                    parent_population,
+                    search_space,
+                    self._rng,
+                    self._swapping_prob,
+                    dominates,
                 )
             else:
-                p1 = p0
+                child = parent_population[0].params
 
-            study._storage.set_trial_system_attr(
-                trial_id, _PARENTS_KEY, [p0._trial_id, p1._trial_id]
-            )
+            # mutation
```

Comment:
As per convention. https://github.com/optuna/optuna/wiki/Coding-Style-Conventions#block-and-inline-comments-are-supposed-to-be-sentences (it's a good read if you haven't already)
```suggestion
            # Mutation.
```

(Actually, the code is quite self explanatory so I even think you can omit this comment altogether)